CMakeLists.txt

```diff
@@ -429,17 +429,19 @@ hpx_option(
   HPX_WITH_DISTRIBUTED_RUNTIME
   BOOL
   "Enable the distributed runtime (default: ON). Turning off the distributed runtime completely disallows the creation and use of components and actions. Turning this option off is experimental!"
-  ON
+  ${HPX_WITH_NETWORKING}
```

Comment:
I would keep these independent, simply because `HPX_WITH_DISTRIBUTED_RUNTIME=OFF` is still bound to have corner cases that are not handled in the configuration and command line options etc. (but please do test it as much as possible). There's a reason I've said that it's experimental here.

There may be a need for renaming these variables, but:
- `HPX_WITH_NETWORKING=OFF` means all code works unchanged (even components and actions), you just can't run HPX with more than one locality.
- `HPX_WITH_DISTRIBUTED_RUNTIME=OFF` means that there are no components, actions, performance counters, logging, and a few other things.

Reply:
All this change does is set the default `HPX_WITH_DISTRIBUTED_RUNTIME` to OFF if you configure the very first time with `cmake -DHPX_WITH_NETWORKING=OFF` from a clean cache of hPX. it just makes the default off if you have disabled networking.

Reply:
I know, and it means that users who have been using `HPX_WITH_NETWORKING=OFF` now get quite a lot of additional features disabled without them noticing (well, they'll notice the configuration errors because they need to disable other things as well when `HPX_WITH_DISTRIBUTED_RUNTIME=OFF`).

Reply:
Well, I don't think we need to have both any more then. Why not just have `HPX_WITH_DISTRIBUTED_RUNTIME` and not bother with the networking option any more. 
If distributed runtime now trumps networking, that's fine. we can just use that.

Reply:
This would need a bit more discussion in that case. I would be happy with that, but I don't know if there are others who still see `HPX_WITH_NETWORKING=OFF` with components and actions as useful. Like I said `HPX_WITH_DISTRIBUTED_RUNTIME=OFF` at the moment also means disabling many other things that a `HPX_WITH_NETWORKING=OFF` user might still want. Naming is clearly unclear here, but I think it might be worth still having the two levels of options, whatever they are called.

Reply:
`HPX_WITH_NETWORKING=OFF` was added principally so that I could work on cholesky a few years back without having the distributed runtime. We were the ones asking for it.

Reply:
As I said, I support this but I think you'd be disappointed at the moment by what's supported by `HPX_WITH_DISTRIBUTED_RUNTIME=OFF`. APEX isn't currently even supported. Some things are easier to fix than others. This tracks things that could potentially be made to work with the local-only runtime: https://github.com/STEllAR-GROUP/hpx/projects/5?add_cards_query=is%3Aopen.